components/prism-tt2.js

```diff
@@ -0,0 +1,61 @@
+(function(Prism) {
+
+	Prism.languages.tt2 = Prism.languages.extend('clike', {
+		comment: {
+			pattern: /#.*|\[%#[\s\S]*?%\]/,
+			lookbehind: true
+		},
+		keyword: /\b(?:BLOCK|CALL|CASE|CATCH|CLEAR|DEBUG|DEFAULT|ELSE|ELSIF|END|FILTER|FINAL|FOREACH|GET|IF|IN|INCLUDE|INSERT|LAST|MACRO|META|NEXT|PERL|PROCESS|RAWPERL|RETURN|SET|STOP|TAGS|THROW|TRY|SWITCH|UNLESS|USE|WHILE|WRAPPER)\b/,
+		punctuation: /[[\]{},()]/
+	});
+
+	delete Prism.languages.tt2['operator'];
+	delete Prism.languages.tt2['variable'];
+	Prism.languages.insertBefore('tt2', 'number', {
+		operator: /(?:=>|==|!=|<=|<|>=|>|=|&&|\|\||\||!|and|or|not)/,
```

Comment:
You could probably combine some patterns here to avoid backtracking.
Also, the parentheses around the whole regexp are not needed.
Finally, wouldn't it be safer to add `\b` assertions around the keyword operators?

Something like `/=[>=]?|!=?|<=?|>=?|&&|\|\|?|\b(?:and|or|not)\b/`?

Reply:
True, that's more efficient and the `\b` also makes the greedy variables unnecessary, see below.

But - sorry for being a smart ass ;) - matching a 58 million character string without any space against `\s` takes about 50 % longer than matching against the more correct `/[\011-\015\040]/`, both in Firefox and Chrome.

Reply:
Hahaha. I take good note of this. And if you really want to revert this change, you might. But you'd have to add a comment above the regexp, specifying what are those characters, something like
`// [\011-\015\040] -> Tab, LF, vertical tab, form feed, CR and space`
or use the character escapes `/[\t\n\v\f\r ]/`. Just so that the meaning is obvious without looking at an ASCII reference table. :wink: